File: java/src/main/java/com/svix/agentic_commerce_protocol/LinkType.java
// This file is @generated
package com.svix.agentic_commerce_protocol;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.agentic_commerce_protocol.Utils.ToQueryParam;

public enum LinkType implements ToQueryParam {
    TERMS_OF_USE("terms_of_use"),
    PRIVACY_POLICY("privacy_policy"),
    SELLER_SHOP_POLICIES("seller_shop_policies");
    private final String value;

    LinkType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/LinkType.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.encodeToJsonElement
import kotlinx.serialization.json.jsonPrimitive

@Serializable
enum class LinkType : ToQueryParam {
    @SerialName("terms_of_use") TERMS_OF_USE,
    @SerialName("privacy_policy") PRIVACY_POLICY,
    @SerialName("seller_shop_policies") SELLER_SHOP_POLICIES;

    override fun toQueryParam() = Json.encodeToJsonElement(this).jsonPrimitive.content
}
